ksql-engine/src/main/java/io/confluent/ksql/physical/PhysicalPlanBuilder.java

```diff
@@ -304,9 +312,9 @@ private void sinkSetUp(final KsqlStructuredDataOutputNode outputNode,
 
   private static String getBareQueryApplicationId(
       final String serviceId,
-      final String transientQueryPrefix
-  ) {
-    return serviceId + transientQueryPrefix + Math.abs(ThreadLocalRandom.current().nextLong());
+      final String transientQueryPrefix,
+      final QueryId queryId) {
+    return serviceId + transientQueryPrefix + queryId;
```

Comment:
With this change, the way we create application IDs for bare queries is the same as for persistent queries (line 251 above), so we could rename this method from `getBareQueryApplicationId` to `getApplicationId` and use it in place of line 251 as well. Not sure whether we prefer this reuse or keeping them separate though.